gap/attr.gi

```diff
@@ -700,6 +700,52 @@ end);
 # returns the vertices (i.e. numbers) of <D> ordered so that there are no
 # edges from <out[j]> to <out[i]> for all <i> greater than <j>.
 
+InstallMethod(UnweightedBellmanFord, "for a digraph by out-neighbours",
+[IsDigraph, IsPosInt],
+function(digraph, source)
+  local distance, n, predecessor, i, inf, u, v, edge, w;
+  n := DigraphNrVertices(digraph);
+  # wouldn't  work for weighted digraphs
+  inf := n + 1;
+  distance := List([1 .. n], x -> 0);
+  predecessor := List([1 .. n], x -> 0);
+  for i in DigraphVertices(digraph) do
+    distance[i] := inf;
+    predecessor[i] := 0;
+  od;
+  distance[source] := 0;
+  for i in [1 .. n - 1] do
+    for edge in DigraphEdges(digraph) do
+      u := edge[1];
+      v := edge[2];
+      # only works for unweighted graphs, w needs to be changed into a variable
```

Comment:
But for this function which is only supposed to do the unweighted version, is there any reason to have this comment or use the variable `w`? Particularly since `w` is not very descriptive.
There is some sort of argument for using a constant `edge_weight := 1` defined near the start of the function, but most people would probably just replace `w` by `1` everywhere.